airbyte-integrations/connectors/source-woocommerce/unit_tests/integration/test_read.py

```diff
@@ -0,0 +1,97 @@
+from .conftest import (
+    coupons_http_calls,
+    customers_http_calls,
+    order_notes_http_calls,
+    orders_empty_last_page,
+    orders_http_calls,
+    payment_gateways_http_calls,
+    product_attribute_terms_http_calls,
+    product_attributes_http_calls,
+    product_categories_http_calls,
+    product_reviews_http_calls,
+    product_shipping_classes_http_calls,
+    product_tags_http_calls,
+    product_variations_http_calls,
+    products_http_calls,
+    refunds_http_calls,
+    shipping_methods_http_calls,
+    shipping_zone_locations_http_calls,
+    shipping_zone_methods_http_calls,
+    shipping_zones_http_calls,
+    system_status_tools_http_calls,
+    tax_classes_http_calls,
+    tax_rates_http_calls,
+)
+
+
+def modified_before() -> str:
+    return "2017-01-29T00:00:00"
+
+
+@freeze_time(modified_before())
+@pytest.mark.parametrize(
+    "stream_name, num_records, http_calls",
+    [
+        # Streams without parent streams
+        ("customers", 2, customers_http_calls()),
+        ("coupons", 2, coupons_http_calls()),
+        ("orders", 2, orders_http_calls()),
+        ("payment_gateways", 4, payment_gateways_http_calls()),
+        ("product_attributes", 2, product_attributes_http_calls()),
+        ("product_categories", 7, product_categories_http_calls()),
+        ("product_reviews", 2, product_reviews_http_calls()),
+        ("products", 2, products_http_calls()),
+        ("product_shipping_classes", 2, product_shipping_classes_http_calls()),
+        ("product_tags", 2, product_tags_http_calls()),
+        ("shipping_methods", 3, shipping_methods_http_calls()),
+        ("shipping_zones", 2, shipping_zones_http_calls()),
+        ("system_status_tools", 9, system_status_tools_http_calls()),
+        ("tax_classes", 3, tax_classes_http_calls()),
+        ("tax_rates", 10, tax_rates_http_calls()),
+        # Streams with parent streams
+        ("order_notes", 6, order_notes_http_calls()),
+        ("product_attribute_terms", 14, product_attribute_terms_http_calls()),
+        ("product_variations", 4, product_variations_http_calls()),
+        ("refunds", 4, refunds_http_calls()),
+        ("shipping_zone_locations", 2, shipping_zone_locations_http_calls()),
+        ("shipping_zone_methods", 4, shipping_zone_methods_http_calls()),
+    ]
+)
+def test_read_simple_endpoints_successfully(stream_name, num_records, http_calls) -> None:
+    """Test basic read for  all streams that don't have parent streams."""
+
+    with requests_mock.Mocker() as m:
+        register_mock_responses(m, http_calls)
+
+        output = read_records(source(), config(), stream_name, SyncMode.full_refresh)
+
+        assert_good_read(output, num_records)
+
+
+@freeze_time("2017-02-10T00:00:00")
+@pytest.mark.parametrize(
+    "stream_name, num_records, http_calls",
+    [("orders", 2, orders_empty_last_page())]
+)
+def test_read_with_multiple_pages_with_empty_last_page_successfully(stream_name, num_records, http_calls) -> None:
```

Comment:
What ensures that we trigger pagination here?

Reply:
The freeze date that goes beyond the first page of one month, and creates second http call. Do you think that is not reliable?

Reply:
One of the benefit from the HttpMocker was that it validated that all the HTTP requests that were mocked were called (at least [when it is used as an annotation](https://github.com/airbytehq/airbyte/blob/148dccda4f0fc5fbec7e3d04805ee76eb1a413ca/airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py#L118-L127) but we have also started to use it through the unittest.TestCase so we should probably add the same logic to [the __exit](https://github.com/airbytehq/airbyte/blob/148dccda4f0fc5fbec7e3d04805ee76eb1a413ca/airbyte-cdk/python/airbyte_cdk/test/mock_http/mocker.py#L44-L45)

Reply:
That's a good point. The intention of this modeling of tests is not to provide all the features but provide a simpler/flexible alternative. For now I added a call-count assertion. Is it OK if we add such features in subsequent improvements?

Reply:
Why not use the HttpMocker if it's already implemented there? Which makes me think that I don't see `from airbyte_cdk.test.utils.http_mocking import register_mock_responses` available [on this branch](https://github.com/airbytehq/airbyte/tree/c9bb9ae2d754751bcb32b68cf3e18af197e69344/airbyte-cdk/python/airbyte_cdk/test/utils)

Reply:
Oh, sorry about the missing file. I'm not using that mocker because it requires the specific `Request` type objects and I wanted to make it simpler. The vanilla http_mocking.Mocker can use simple requests and regexes that make it easier to match things.